code/modules/recycling/disposal/holder.dm

```diff
@@ -90,6 +91,31 @@
 	current_pipe = null
 	last_pipe = null
 	active = FALSE
+	for(var/mob/living/piperider in contents)
+		to_chat(piperider, span_notice("Your movement has slowed to a stop. If you tried, you could probably <b>struggle</b> free."))
+
+
+/obj/structure/disposalholder/proc/struggle(mob/living/escapee)
+	SIGNAL_HANDLER
+	if(escapee.loc != src)
+		UnregisterSignal(escapee, COMSIG_LIVING_RESIST)
+		return //Somehow they got out without telling us
+	if(!istype(loc, /obj/structure/disposalpipe))
+		return //Somehow we're not in a pipe, shits probably fucked
+	var/obj/structure/disposalpipe/transport_cylinder = loc
+	if(active)
+		to_chat(escapee, span_danger("You slide past [loc] and are unable to keep your grip!"))
+		return
+	if(src in escapee.do_afters)
+		return //already trying to escape
+	to_chat(escapee, span_warning("You push against the thin pipe walls..."))
+	playsound(loc, 'sound/machines/airlock_alien_prying.ogg', vol = 30, vary = FALSE, extrarange = 3) //yeah I know but at least it sounds like metal being bent.
+	if(!do_after(escapee, 20 SECONDS, get_turf(loc)))
+		return
+	for(var/mob/living/jailbird in contents)
+		jailbird.apply_damage(rand(5,15), damagetype = BRUTE)
```

Comment:
![image](https://user-images.githubusercontent.com/37497534/198864279-d4850071-cd04-4c80-b119-2ecb25cf4711.png)

This is all mobs, for consistency mostly. It seems weird to me that only the one breaking free would take damage, when the damage is mostly a representation of the sudden pressure change and possibly sharp edges of the broken pipe.